### PR TITLE
Cppyy bug

### PR DESCRIPTION
**Pull request by faze-geek**
# Description

Please include a summary of changes, motivation and context for this PR.

Fixes  https://github.com/compiler-research/CppInterOp/pull/365

## Type of change

Please tick all options which are relevant.

- [x] Bug fix
- [ ] New feature
- [ ] Requires documentation updates

## Checklist

- [x] I have read the contribution guide recently

### CODE REVIEW

**Comment by faze-geek** [comment]
@Vipul-Cariappa All tests pass locally, though I can see a bunch of clang - tidy suggestions. Can you have a look ?

File: lib/Interpreter/CppInterOp.cpp [hunk 1]
```diff
@@ -1212,12 +1212,20 @@ namespace Cpp {
     return 0;
   }
 
-  TCppType_t GetVariableType(TCppScope_t var)
-  {
-    auto D = (Decl *) var;
+  TCppType_t GetVariableType(TCppScope_t var) {
+    auto D = (Decl*)var;
```
Comment (Vipul-Cariappa) [inline lib/Interpreter/CppInterOp.cpp:1216]: You should be able to satisfy clang-tidy with this

```suggestion
    auto *D = static_cast<Decl*>(var);
```

**Review by Vipul-Cariappa — APPROVED** [review]
Looks good to me.
Enables `test03_stllike_preinc` at cppyy.

We should consider "squash and merge" and update the comment message to something more meaningful.
I might be a contributor for the poor initial commit message.

@vgvassilev if you can also approve this. I will go ahead and merge.

File: lib/Interpreter/CppInterOp.cpp [hunk 2]
```diff
@@ -1212,12 +1212,20 @@ namespace Cpp {
     return 0;
   }
 
-  TCppType_t GetVariableType(TCppScope_t var)
-  {
-    auto D = (Decl *) var;
+  TCppType_t GetVariableType(TCppScope_t var) {
+    auto *D = (Decl*)var;
```
Comment (vgvassilev) [inline lib/Interpreter/CppInterOp.cpp:1216]:
```suggestion
    auto *D = static_cast<Decl*>(var);
```

**Review by vgvassilev — APPROVED** [review]
LGTM if the bots are happy.

File: lib/Interpreter/CppInterOp.cpp [hunk 3]
```diff
@@ -1212,12 +1212,20 @@ namespace Cpp {
     return 0;
   }
 
-  TCppType_t GetVariableType(TCppScope_t var)
-  {
-    auto D = (Decl *) var;
+  TCppType_t GetVariableType(TCppScope_t var) {
+    auto *D = static_cast<Decl*>(var);
```
Comment (Vipul-Cariappa) [inline lib/Interpreter/CppInterOp.cpp:1216]:
```suggestion
    auto* D = static_cast<Decl*>(var);
```

**Pull request merged**